src/libostree/ostree-sysroot-deploy.c

```diff
@@ -1458,20 +1458,30 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
     ostree_bootconfig_parser_set (bootconfig, "linux", boot_relpath);
   }
 
+  val = ostree_bootconfig_parser_get (bootconfig, "options");
+  kargs = _ostree_kernel_args_from_string (val);
+
   if (dest_initramfs_name)
     {
       g_autofree char * boot_relpath = g_strconcat ("/", bootcsumdir, "/", dest_initramfs_name, NULL);
       ostree_bootconfig_parser_set (bootconfig, "initrd", boot_relpath);
     }
-
-  val = ostree_bootconfig_parser_get (bootconfig, "options");
+  else
+    {
+      g_autofree char *ostree_prepare_root = NULL;
+      ostree_prepare_root = g_strdup_printf ("init=/ostree/boot.%d/%s/%s/%d/usr/lib/ostree-boot/ostree-prepare-root /",
```

Comment:
For a better backward compatibility I suggest: `#ifdef  BUILDOPT_STATIC_PREPARE_ROOT`

Now when ostree is build with static `ostree-prepare-root` users

- can still use initramfs code path, if `initramfs-*` found (by using the static version inside of initramfs)
- when `initramfs-*` not found, append` init=` kernel arg.

and if there are any users out there that were using OSTree without initramfs already (by hacking kernel sources to include the ostree-prepare-root logic), they can continue doing that by not compiling `ostree-prepare-root` statically (`BUILDOPT_STATIC_PREPARE_ROOT` not defined.)